pkg/actuators/machine/instances.go

```diff
@@ -577,6 +577,10 @@ func getInstanceMarketOptionsRequest(providerConfig *machinev1beta1.AWSMachinePr
 		return nil, errors.New("can't create spot capacity-blocks, remove spot market request")
 	}
 
+	if (providerConfig.MarketType == machinev1beta1.MarketTypeSpot || providerConfig.SpotMarketOptions != nil) && providerConfig.CapacityReservationID != "" {
+		return nil, errors.New("can't create spot capacity-blocks, remove capacityReservationID or remove spot requests")
```

Comment:
Let's reword this message like we did upstream
```suggestion
		return nil, errors.New("unable to generate marketOptions for spot instance, capacityReservationID is incompatible with marketType spot and spotMarketOptions")
```